openmc/model/surface_composite.py

```diff
@@ -51,6 +52,138 @@ def __neg__(self):
+    are parallel to the first basis axis, two surfaces are parallel
+    to the second basis axis, and the remaining four surfaces intersect both
+    basis axes at 45 degree angles.
+
+    This class acts as a proper surface, meaning that unary `+` and `-`
+    operators applied to it will produce a half-space. The negative side is
+    defined to be the region inside of the octogonal prism.
+
+    Parameters
+    ----------
+    center : iterable of float
+        Coordinate for the central axis of the octagon in the
+        (y, z), (z, x), or (x, y) basis.
+    r1 : float
+        Half-width of octagon across its basis axis-parallel sides in units
+        of cm. Must be less than :math:`r_2\sqrt{2}`.
+    r2 : float
+        Half-width of octagon across its basis axis intersecting sides in
+        units of cm. Must be less than than :math:`r_1\sqrt{2}`.
+    axis : {'x', 'y', 'z'}
+        Central axis of octagon. Defaults to 'z'
+    **kwargs
+        Keyword arguments passed to underlying plane classes
+
+    Attributes
+    ----------
+    top : openmc.ZPlane, openmc.XPlane, or openmc.YPlane
+        Top planar surface of octagon
+    bottom : openmc.ZPlane, openmc.XPlane, or openmc.YPlane
+        Bottom planar surface of octagon
+    right: openmc.YPlane, openmc.ZPlane, or openmc.XPlane
+        Right planar surface of octagon
+    left: openmc.YPlane, openmc.ZPlane, or openmc.XPlane
+        Left planar surface of octagon
+    upper_right : openmc.Plane
+        Upper right planar surface of octagon
+    lower_right : openmc.Plane
+        Lower right planar surface of octagon
+    lower_left : openmc.Plane
+        Lower left planar surface of octagon
+    upper_left : openmc.Plane
+        Upper left planar surface of octagon
+
+    """
+
+    _surface_names = ('top', 'bottom',
+                      'upper_right', 'lower_left',
+                      'right', 'left',
+                      'lower_right', 'upper_left')
+
+    def __init__(self, center, r1, r2, axis='z', **kwargs):
+        c1, c2 = center
+
+        # Coords for axis-perpendicular planes
+        ctop = c1 + r1
+        cbottom = c1 - r1
+
+        cright = c2 + r1
+        cleft = c2 - r1
+
+        # Side lengths
+        if r2 > r1 * sqrt(2):
+            raise ValueError(f'r2 is greater than sqrt(2) * r1. Octagon' + \
+                             ' may be erroneous.')
+        if r1 > r2 * sqrt(2):
+            raise ValueError(f'r1 is greater than sqrt(2) * r2. Octagon' + \
+                             ' may be erroneous.')
+
+        L_perp_ax1 = (r2 * sqrt(2) - r1) * 2
+        L_perp_ax2 = (r1 * sqrt(2) - r2) * 2
+
+        # Coords for quadrant planes
+        p1_ur = array([L_perp_ax1/2, r1, 0.])
+        p2_ur = array([r1, L_perp_ax1/2, 0.])
+        p3_ur = array([r1, L_perp_ax1/2, 1.])
+
+        p1_lr = array([r1, -L_perp_ax1/2, 0.])
+        p2_lr = array([L_perp_ax1/2, -r1, 0.])
+        p3_lr = array([L_perp_ax1/2, -r1, 1.])
```

Comment:
If we're always using `L_prep_ax1/2` might as well just have the variable be half of what it is now (eliminating all the `/2`)